client/src/modules/cohort/model/reducer.js

```diff
@@ -21,24 +21,60 @@ const cohorts = (state = {}, action) => {
         [action.payload.cohortId]: updatedCohort
       };
     }
+    case CohortActionTypes.ARCHIVE_SUCCESS: {
+      const { cohortId: _id, isArchived } = action.payload;
+      const { name } = state[_id];
+      const archivedCohort = { _id, isArchived, name };
+      return {
+        ...state,
+        [_id]: archivedCohort
```

Comment:
consider changin this reducer to 
```
const previousCohort = state[_id];

return {
...state,
[_id]: {...previousCohort, isArchved}
```

Reply:
I think we can't do this because we want to remove all data we don't need except { _id, isArchived, name }, we remove permissions, description... Your proposal is much more cleaner but don't remove unnecessary data...